crates/sui-types/src/crypto.rs

```diff
@@ -603,6 +607,13 @@ where
     <KP as KeypairTraits>::PubKey: SuiPublicKey,
 {
     let priv_length = <KP as KeypairTraits>::PrivKey::LENGTH;
+    if bytes.len() != priv_length {
```

Comment:
Interestingly, if you see below, we read the private key from the key pair bytes as `&bytes[..priv_length]`. Your test reveals that we store keypairs as `priv||pub` Vs just `priv`, which is a useful outcome and IMO we should fix (and that's the reason your tests currently fail as it expects 32B but it got 64B due to the keypair concatenation).

It's a good hygiene (you see what happened in the ed25519 key mismatch attack, right?) to store the secret only.

Reply:
This is only used in tests, so I've fixed this to read according to both key lenghts.